### ISSUE TITLE
Installing inside docker in a raspberry pi with files in external ext4 hdd

### ISSUE THREAD

**Issue by miqmago** [LibreSign/libresign#1685]
**Describe the bug**
I'm getting `bash: ./java: Permission denied`

**To Reproduce**
I've installed all the libraries:

```sh
# from outside docker
sudo git clone https://github.com/codemasher/php-ext-xz.git
sudo chown www-data:www-data php-ext-xz/ -R

# from inside docker
docker exec -it -u 0 nextcloud_app bash
apt-get update
apt-get install git liblzma-dev -y
cd ..
mv html/php-ext-xz/ .
cd php-ext-xz
phpize
./configure
make
make install
echo "extension=xz.so" >> /usr/local/etc/php/conf.d/nextcloud.ini
exit

docker exec -u www-data nextcloud_app php /var/www/html/occ libresign:install --all
docker exec -u www-data nextcloud_app bash -c /var/www/html/data/appdata_xxxxxxxxx/libresign/java/jdk-17.0.5+8-jre/bin/java
```

Always getting 
```
bash: line 1: /var/www/html/data/appdata_xxxxxxxxx/libresign/java/jdk-17.0.5+8-jre/bin/java: Permission denied
```

**Expected behavior**
Java should be executable

**Screenshots**
<img width="940" alt="Captura de pantalla 2023-04-30 a les 23 23 26" src="https://user-images.githubusercontent.com/1849488/235376837-65c982c6-b5ef-4bbc-a3b7-d240d968309e.png">

**Environment information (please complete the following information):**
 - OS: Docker (Linux 6.1.21-v8+ aarch64 - Raspberry Pi 4 Model B Rev 1.4 (4 cores))
 - Browser n/a
 - LibreSign Version 7.1.1 - 2023-04-12
 - Nextcloud Server Version 26.0.1
 - Logs n/a

**Comment by NicolasDequeant** [comment]
Hi, xz issue/ raspberry : 

Sounds like something to do with this thread [](https://github.com/LibreSign/libresign/issues/1675)

Regards,

**Comment by miqmago** [comment]
But the problem has to do with java... I've tried to execute the binnary also outside docker and the problem is the same: `Permission denied`.

**Comment by NicolasDequeant** [comment]
Not a specialist but I would check the ownership of the data folder of NC.

**Comment by miqmago** [comment]
I've tried to change permissions too, but the result is the same. Permissions are:

```
root@raspberrypi:/home/pi/nextcloud/mnt/data/appdata_xxxxxxxxx/libresign/java/jdk-17.0.5+8-jre/bin# ls -la
total 368
drwxr-xr-x 2 www-data www-data  4096 Apr 30 23:05 .
drwxr-xr-x 6 www-data www-data  4096 Apr 30 23:05 ..
-rwxr-xr-x 1 www-data www-data 70704 Oct 19  2022 java
-rwxr-xr-x 1 www-data www-data 70720 Oct 19  2022 jfr
-rwxr-xr-x 1 www-data www-data 70744 Oct 19  2022 jrunscript
-rwxr-xr-x 1 www-data www-data 70720 Oct 19  2022 keytool
-rwxr-xr-x 1 www-data www-data 70728 Oct 19  2022 rmiregistry
```

**Comment by miqmago** [comment]
[**SOLVED**] but any explanation would be really appreciated!

This is quite weird... My installation of nextcloud has an external hard drive mounted on `/home/pi/nextcloud/mnt`, so all files of nextcloud are saved on an external 4TB hdd.

In my `docker-compose.yml` I've defined a volume like this:

```yml
    volumes:
      - ./mnt:/var/www/html
```

Seems that docker cannot execute java saved on hdd so I've entered inside docker and moved java to `/usr/local/lib`.
Then I've created a symnlink on `libresign` folder and now it works fine:

```bash
docker exec -it -u0 nextcloud_app bash
cd appdata_xxxxxxxxx/libresign/java
mv jdk-17.0.5+8-jre /usr/local/lib
ln -s /usr/local/lib/jdk-17.0.5+8-jre jdk-17.0.5+8-jre
```
